code/espurna/scheduler.cpp

```diff
@@ -60,6 +62,96 @@ constexpr bool restoreLast() {
 
 // -----------------------------------------------------------------------------
 
+#if API_SUPPORT
+
+// UTILS
+
+template <typename V = const char*>
+bool _setJsonVariant(const SettingsKey& k, const JsonVariant& v) {
+    if (!v.is<V>()) {
+        return false;
+    }
+    return setSetting(k, std::move(v.as<V>()));
+}
+
+template <>
+bool _setJsonVariant<const char*>(const SettingsKey& k, const JsonVariant& v) {
+    const auto& tmp = v.as<const char*>();
+    if (tmp == nullptr) {
+        return false;
+    }
+    return setSetting(k, std::move(tmp));
+}
+
+template <typename T = const char*, typename ObjKey>
+bool _setJsonKey(const SettingsKey& k, const JsonObject& o, const ObjKey& key) {
+    if (!o.containsKey(key)) {
+        return false;
+    }
+    return _setJsonVariant<T>(k, o[key]);
+}
+
+// taken from relay.cpp
```

Comment:
note of the `tryParseId` from the utils, and the `scheduler::build::max()` (and the rest of the ::build namespace)

Reply:
is `scheduler::build::max()` new?

Reply:
yes. this was part of another update to push constant definitions into const/constexpr functions and variables that came with the... sizable update 2d ago